ports/geckolib/values.rs

```diff
@@ -159,14 +159,16 @@ impl GeckoStyleCoordConvertible for LengthOrPercentage {
     fn to_gecko_style_coord(&self, unit: &mut nsStyleUnit, union: &mut nsStyleUnion) {
         match *self {
             LengthOrPercentage::Length(au) => {
+                unsafe { union.reset(unit) };
```

Comment:
Either way sounds OK to me, though if you pull the `reset()` call out please assert in `set_calc_value` that the unit is `eStyleUnit_Null`.

Alternatively, you could have a `Gecko_NewStyleCoordCalc` function that just creates the new refcounted `StyleCoordCalc` value and returns it, and `to_gecko_style_coord` here can set the unit and assign the `StyleCoordCalc` pointer into `union.mPointer`.  That would make all the branches of the match statement here line up.

Regardless, I think we'll need to make `StyleCoordValue` support thread-safe refcounting.  And we probably want to change all setter methods in `impl StyleCoordHelpers for nsStyleCoord` to do the right thing if the current (to be overwritten) value is a calc, and for `copy_from` to work if `other` is a calc too.